### PR TITLE
Refactor/modularity separation of the VisitService from the ClinicService

### PR DESCRIPTION
**Pull request by Depdx**
As per code review. Some adjustment were needed to have the visit feature be a module on it's own. To do so, the `VisitService` class was added and visit feature was removed from the `ClinicService`. Hence, making the visit feature modular.

### CODE REVIEW

File: src/main/java/org/springframework/samples/petclinic/service/VisitService.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import org.springframework.samples.petclinic.model.Visit;
+
+
+/**
+ * Mostly used as a facade for the visit controller
+ *
+ * @author Nicolas Depelteau
+ */
+public interface VisitService {
+
```
Comment (quentinguidee) [inline src/main/java/org/springframework/samples/petclinic/service/VisitService.java]:
```suggestion
```

File: src/main/java/org/springframework/samples/petclinic/service/VisitServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    }
+
+	@Override
+	@Transactional(readOnly = true)
+	public Visit findVisitById(int visitId) throws DataAccessException {
+		Visit visit = null;
+		try {
+			visit = visitRepository.findById(visitId);
+		} catch (ObjectRetrievalFailureException|EmptyResultDataAccessException e) {
+		// just ignore not found exceptions for Jdbc/Jpa realization
```
Comment (quentinguidee) [inline src/main/java/org/springframework/samples/petclinic/service/VisitServiceImpl.java]:
```suggestion
			// just ignore not found exceptions for Jdbc/Jpa realization
```

File: src/main/java/org/springframework/samples/petclinic/service/VisitServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+	@Transactional
+	public void deleteVisit(Visit visit) throws DataAccessException {
+		visitRepository.delete(visit);
+	}
+
+	@Override
+	@Transactional
+	public void saveVisit(Visit visit) throws DataAccessException {
+		visitRepository.save(visit);
+
```
Comment (quentinguidee) [inline src/main/java/org/springframework/samples/petclinic/service/VisitServiceImpl.java]:
```suggestion
```

File: src/main/java/org/springframework/samples/petclinic/util/CallMonitoringAspect.java [hunk 4]
```diff
@@ -59,6 +59,7 @@ public void reset() {
         this.callCount = 0;
         this.accumulatedCallTime = 0;
     }
+
```
Comment (quentinguidee) [inline src/main/java/org/springframework/samples/petclinic/util/CallMonitoringAspect.java]:
```suggestion
```

**Review by quentinguidee — CHANGES_REQUESTED** [review]
Everything seems good; just highlighted some formatting issues

**Comment by Jamesley55** [comment]
LGTM!

**Pull request merged**